test/indexOf.js

```diff
@@ -83,4 +83,20 @@ describe('indexOf', function() {
     var curried = R.indexOf(3);
     eq(curried(list), 2);
   });
+
+  it('finds function, compared by identity', function() {
+    var f = function() {};
+    var g = function() {};
+    var list = [g, f];
```

Comment:
You could make this `[g, f, g, f]` (as you do in the equivalent test for `R.lastIndexOf`) to assert that the first occurrence is matched.